app/src/androidTest/java/ch/hikemate/app/endtoend/EndToEndTest.kt

```diff
@@ -1,40 +1,77 @@
 package ch.hikemate.app.endtoend
 
+import android.content.Context
 import androidx.compose.ui.test.assertHasClickAction
 import androidx.compose.ui.test.assertIsDisplayed
 import androidx.compose.ui.test.assertIsNotDisplayed
 import androidx.compose.ui.test.junit4.createAndroidComposeRule
 import androidx.compose.ui.test.onNodeWithTag
 import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performTextInput
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
 import ch.hikemate.app.MainActivity
 import ch.hikemate.app.ui.auth.SignInScreen
+import ch.hikemate.app.ui.auth.SignInWithEmailScreen
 import ch.hikemate.app.ui.map.MapScreen.TEST_TAG_MAP
 import ch.hikemate.app.ui.navigation.LIST_TOP_LEVEL_DESTINATIONS
+import ch.hikemate.app.ui.navigation.Screen
 import ch.hikemate.app.ui.navigation.Screen.PROFILE
 import ch.hikemate.app.ui.navigation.TEST_TAG_DRAWER_CLOSE_BUTTON
 import ch.hikemate.app.ui.navigation.TEST_TAG_DRAWER_CONTENT
 import ch.hikemate.app.ui.navigation.TEST_TAG_DRAWER_ITEM_PREFIX
 import ch.hikemate.app.ui.navigation.TEST_TAG_SIDEBAR_BUTTON
 import ch.hikemate.app.ui.navigation.TopLevelDestinations
 import ch.hikemate.app.ui.saved.TEST_TAG_SAVED_HIKES_SECTION_CONTAINER
+import com.google.firebase.FirebaseApp
+import com.google.firebase.auth.FirebaseAuth
+import junit.framework.TestCase
+import org.junit.After
+import org.junit.Before
 import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
 
-// @RunWith(AndroidJUnit4::class)
-// TODO: Change it to "EndToEndTest: TestCase()"
-class EndToEndTest {
+@RunWith(AndroidJUnit4::class)
+class EndToEndTest : TestCase() {
   @get:Rule val composeTestRule = createAndroidComposeRule<MainActivity>()
+  private val auth = FirebaseAuth.getInstance()
+  private val email = "end2endTest@gmail.com"
+  private val password = "password"
 
-  // @Test
+  @Before
+  fun setupFirebase() {
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    FirebaseApp.initializeApp(context)
+
+    auth.createUserWithEmailAndPassword(email, password)
+    auth.signOut()
+  }
+
+  @After
+  fun deleteUser() {
+    auth.currentUser?.delete()
+    auth.signOut()
```

Comment:
Question: Is signOut necessary after deleting the user?

Reply:
Deleting the user should signing him out. However, when testing, it could be unreliable so I added this sanity check here.

Reply:
Could you add a small comment about it? So that if people don't read this PR, they know why it has been done this way

Reply:
Sure